### PR TITLE
[WIP] Expand cluster with given capacity in Gi format 

### PR DESCRIPTION
**Pull request by annagitel**
- Moved add_storage_capacity function to storageCluster class. 
- Changed add capacity test

Work on the PR is still in progress. Wasn't tested yet on a cluster. 
Please review the storageCluster file

### CODE REVIEW

File: ocs_ci/ocs/resources/storage_cluster.py [hunk 1]
```diff
@@ -23,3 +28,77 @@ def __init__(self, resource_name="", *args, **kwargs):
...
+
+    ocp = OCP(namespace=defaults.ROOK_CLUSTER_NAMESPACE, kind=constants.STORAGECLUSTER)
+    osd_capacity = parse_size_to_int(get_storage_cluster()['spec']['storageDeviceSets']
+                                        [0]['dataPVCTemplate']['spec']
+                                        ['resources']['resources']['storage'])
+
+    osd_replicas = parse_size_to_int(get_storage_cluster()['spec']['storageDeviceSets'][0]['replicas'])
+    worker_nodes = len(helpers.get_worker_nodes())
+    current_osd_count = len(pod.get_osd_pods())
+    available_osd_number = worker_nodes * 3 - current_osd_count
```
Comment (hnallurv) [inline ocs_ci/ocs/resources/storage_cluster.py]: replace worker_nodes * 3 - current_osd_count with "(worker_nodes * 3) - current_osd_count" to improve readability.

File: ocs_ci/ocs/resources/storage_cluster.py [hunk 2]
```diff
@@ -23,3 +28,77 @@ def __init__(self, resource_name="", *args, **kwargs):
...
+                params=f'[{{"op": "replace", "path": "/spec/storageDeviceSets/0/count", '
+                       f'"value":{capacity / osd_capacity * osd_replicas}}}]'
+            )
+
+            ocp.patch(
+                resource_name=sc['metadata']['name'],
+                params=f'[{{"op": "replace", "path": "/spec/storageDeviceSets/0/dataPVCTemplate/spec/resources'
+                       f'/requests/storage", "value":{capacity}}}] '
+            )
+            return True
```
Comment (hnallurv) [inline ocs_ci/ocs/resources/storage_cluster.py]: Before returning true, please make sure the newly added OSD pods are  in 'running' state.

File: ocs_ci/ocs/resources/storage_cluster.py [hunk 3]
```diff
@@ -23,3 +28,77 @@ def __init__(self, resource_name="", *args, **kwargs):
...
+    available_osd_number = worker_nodes * 3 - current_osd_count
+    capacity = parse_size_to_int(capacity_string)
+
+    if capacity % osd_capacity == 0:
+        if capacity / osd_capacity * osd_replicas <= available_osd_number:
+            sc = get_storage_cluster()
+            ocp.patch(
+                resource_name=sc['metadata']['name'],
+                params=f'[{{"op": "replace", "path": "/spec/storageDeviceSets/0/count", '
+                       f'"value":{capacity / osd_capacity * osd_replicas}}}]'
```
Comment (hnallurv) [inline ocs_ci/ocs/resources/storage_cluster.py]: The count here is not osd count. It's storageDeviceSets count. I can be calculated as : 
new count for storageDeviceSets = (capacity reqested / osd capacity ) + exiting count storageDeviceSets 

Please check and change the calculation

I was able to get 3 340GiB OSDs by just changing the "storageDeviceSets -> count" to 2 from 1 in "Web Console -> Installed Operators -> ocs-operator.v0.0.287 -> StorageCluster Details -> YAML"

File: ocs_ci/ocs/resources/storage_cluster.py [hunk 4]
```diff
@@ -23,3 +28,77 @@ def __init__(self, resource_name="", *args, **kwargs):
...
+            )
+
+            ocp.patch(
+                resource_name=sc['metadata']['name'],
+                params=f'[{{"op": "replace", "path": "/spec/storageDeviceSets/0/dataPVCTemplate/spec/resources'
+                       f'/requests/storage", "value":{capacity}}}] '
```
Comment (hnallurv) [inline ocs_ci/ocs/resources/storage_cluster.py]: This is not needed.

File: tests/manage/z_cluster/cluster_expansion/test_add_capacity.py [hunk 5]
```diff
@@ -17,35 +20,31 @@ class TestAddCapacity(ManageTest):
...
         Test to add variable capacity to the OSD cluster while IOs running
 
         Args:
-        node_multiplier: the number of OSD to add per worker node
-        capacity: the storage capacity of each OSD
+            capacity (String):the storage capacity of each OSD
+            expected (Boolean): Expected boolean statment from add_capacity function
         """
         dt = config.ENV_DATA['deployment_type']
         if dt == 'ipi':
```
Comment (hnallurv) [inline tests/manage/z_cluster/cluster_expansion/test_add_capacity.py]: I don't think it matters to check for ipi and upi here. Capacity expansion is same for both. Node expansion differs.

File: ocs_ci/ocs/resources/storage_cluster.py (hunk 1 above)
Comment (annagitel) [reply ocs_ci/ocs/resources/storage_cluster.py]: Done

File: ocs_ci/ocs/resources/storage_cluster.py (hunk 4 above)
Comment (annagitel) [reply ocs_ci/ocs/resources/storage_cluster.py]: Done

File: tests/manage/z_cluster/cluster_expansion/test_add_capacity.py (hunk 5 above)
Comment (annagitel) [reply tests/manage/z_cluster/cluster_expansion/test_add_capacity.py]: Done

**Pull request closed without merging**